restcomm/restcomm.dao/src/main/java/org/mobicents/servlet/restcomm/dao/mybatis/MybatisGeolocationDao.java

```diff
@@ -0,0 +1,213 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.mobicents.servlet.restcomm.dao.mybatis;
+
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readDateTime;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readInteger;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readSid;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readString;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readUri;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeDateTime;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeSid;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.writeUri;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readGeolocationType;
+import static org.mobicents.servlet.restcomm.dao.DaoUtils.readLong;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.joda.time.DateTime;
+import org.mobicents.servlet.restcomm.dao.GeolocationDao;
+import org.mobicents.servlet.restcomm.entities.Geolocation;
+import org.mobicents.servlet.restcomm.entities.Sid;
+
+/**
+ * @author fernando.mendioroz@telestax.com (Fernando Mendioroz)
+ *
+ */
+public class MybatisGeolocationDao implements GeolocationDao {
+
+    private static final String namespace = "org.mobicents.servlet.sip.restcomm.dao.GeolocationDao.";
+    private final SqlSessionFactory sessions;
+
+    public MybatisGeolocationDao(final SqlSessionFactory sessions) {
+        super();
+        this.sessions = sessions;
+    }
+
+    @Override
+    public void addGeolocation(Geolocation gl) {
+        final SqlSession session = sessions.openSession();
+        try {
+            session.insert(namespace + "addGeolocation", toMap(gl));
+            session.commit();
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public Geolocation getGeolocation(Sid sid) {
+        return getGeolocation(namespace + "getGeolocation", sid.toString());
+    }
+
+    private Geolocation getGeolocation(final String selector, final String parameter) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final Map<String, Object> result = session.selectOne(selector, parameter);
+            if (result != null) {
+                return toGeolocation(result);
+            } else {
+                return null;
+            }
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public List<Geolocation> getGeolocations(Sid accountSid) {
+        final SqlSession session = sessions.openSession();
+        try {
+            final List<Map<String, Object>> results = session.selectList(namespace + "getGeolocations", accountSid.toString());
```

Comment:
do we need to call toString() here. Also toString is better to be kept as diagnosis purposes like logging, maybe better to create a method that returns whatever you need..